File: test-junit/src/test/java/io/micronaut/test/junit/MathController.java

package io.micronaut.test.junit;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

@Controller("/math")
public class MathController {
    MathService mathService;

    MathController(MathService mathService) {
        this.mathService = mathService;
    }

    @Get(uri = "/compute/{number}", processes = MediaType.TEXT_PLAIN)
    String compute(Integer number) {
        return String.valueOf(mathService.compute(number));
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/MathController.py
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get

from .MathService import MathService


@Controller("/math")
class MathController:

    def __init__(self, math_service: MathService):
        self.math_service = math_service

    @Get(uri="/compute/{number}", processes=MediaType.TEXT_PLAIN)
    def compute(self, number: int) -> str:
        return str(self.math_service.compute(number))
